include/config/battle.h

```diff
@@ -164,6 +164,14 @@
 #define VAR_TERRAIN                 0     // If this var has a value, assigning a STATUS_FIELD_xx_TERRAIN to it before battle causes the battle to start with that terrain active
 #define B_VAR_WILD_AI_FLAGS         0     // If not 0, you can use this var to add to default wild AI flags. NOT usable with flags above (1 << 15)
 
+// Sky Battles
+#define B_FLAG_SKY_BATTLE           0           // If this flag has a value, the player will be able to engage in scripted Sky Battles.
+#define B_VAR_SKY_BATTLE            0           // If this var has a value, the game will remember the positions of Pokémon used in Sky Battles.
+
+#define B_SKY_BATTLE_STRICT_ELIGIBILITY FALSE //If TRUE, Sky Battles will use the eligibility from Pokémon XY. If FALSE, all Flying-types or Pokémon with Levitate are allowed.
```

Comment:
Personally, I'm not a fan of distinguishing between strict and non-strict variants. I would just get rid of these and make the mechanics the same as in XY. And for mons from gens7/8/9 I would follow the same logic as gen6 games. Though, you could ask other people to see what theirs opinion on this matter is.

Reply:
XY "mechanics" are "did Creatures Inc make a levitating or flying model for this Pokémon". 
the list truly doesn't make any sense. :( I **think** most people would agree that a sane requirement is "Flying or Levitate". I've included both to give developers the choice (for example, an XY demake, which I know a decomp is currently in progress for)

Reply:
I think it's because of the arbitrary restriction of the models that I actively dislike Sky Battles. Pidgey unable to participate while Pidgeotto and Pidgeot can is asinine to me, it's literally because Pidgey is standing while the other ones are not, so at least I welcome the option to remove those restrictions, as we can't guarantee that our sprites match the XY poses.

Reply:
I agree with Psf and Eduardo here. I think letting Flying types and Pokémon with Levitate participate in Sky Battles regardless of their sprite and pose is an okay compromise to make. Otherwise no one would want to use them, and those who can read and edit code would likely end up making a similar change anyway 😆 

The toggle is set to `FALSE` by default anyway, so Sky Battles would work just like in XY by default, this is just a nice extra for people making projects.